### DISCUSSION TITLE
Sharing states between domains?

### DISCUSSION THREAD

**Discussion by nicnocquee** [pointfreeco/swift-composable-architecture#623] [Q&A] [answered]
Hello,

I wonder if anyone can give suggestions on how to share a state between domains?

From this [episode-code-samples](https://github.com/pointfreeco/episode-code-samples/blob/f5cd3b4916d4782030eda09815c6577e9faa3c25/0147-derived-behavior-pt2/DerivedBehavior/DerivedBehavior/TcaContentView.swift),

```swift
struct AppState: Equatable {
  var count = 0
  var favorites: Set<Int> = []

  var counter: CounterState {
    get {
      .init(count: self.count, favorites: self.favorites)
    }
    set {
      self.count = newValue.count
      self.favorites = newValue.favorites
    }
  }
  var profile: ProfileState {
    get {
      .init(favorites: self.favorites)
    }
    set {
      self.favorites = newValue.favorites
    }
  }
}
```

the `favourites` is duplicated in three places: AppState, CounterState, and ProfileState. Is this really the recommended way?

**Comment by stephencelis** [comment, marked as answer]
While it may look like `favorites` is duplicated 3 places, `AppState.counter` and `AppState.profile` are computed properties that use `AppState.favorites` under the hood. This means there's really only a single `favorites` field in state, and not 3, so there's no way for `AppState.favorites`, `CounterState.favorites`, or `ProfileState.favorites` to ever fall out of sync. They all use the same single field as a backing store under the hood.

There is boilerplate involved in managing these computed sub-sets of state, but it's typically the same amount of code (or less) than equivalent code that manually coordinates state between components.

**Reply by nicnocquee** [reply]
I see. Thanks for the clarification! Love TCA, btw 👍
